### ISSUE TITLE
Unity for Android: PlatformLocationProvider not providing GPS or WiFi on some Android devices 

### ISSUE THREAD

**Issue by TobiasPott** [Azure/azure-spatial-anchors-samples#88]
Hi,

The PlatformLocationProvider does not provide GPS and/or WiFi information the example can use to save new anchors with. The status indicators keep displaying a yellow color although any device I've tested it on has location services enabled and permissions granted.
I've been working around this issue for the last few hours and extended the devices I've checked it on to any available Android device I have at hand here in office and experience different behaviour I don't know any solution to.
I'm using the current version (2.0) of the example repository (and as a second test a modified version for our actual app we are developing) and the behaviour is identical regardless of the app running.

The status gets updated and signalled as 'ok' on my Google Pixel 2 XL (Android 10) and on a colleagues Samsung Galaxy S8. On our development/testing devices they stay yellow indicating a 'missing provider' according to the UpdateSensorState component from the examples. Our Test-Devices are Motorola Moto G5S Plus (Android 8.1) and Moto G7 Play (Android 9.0), which both are ARCore compatible and do work with the other parts of the examples (basic, nearby and local shared anchors).
All devices have the Google Play Services for AR installed in version 1.13.xx

The project is build on a Windows machine from Unity 2019.2.11. According to the examples app nothing was modified from the state the repository has.

As @reneschulte had some issues with GPS and WiFi in #87 I wondered if there might be a connection and tried disabling GPS or WiFi data on the sensor instance inside my app. This did not make any difference.

I actually don't know what the problem might be and how to tackle it. And it might be helpful if anyone else could confirm this issue with more medium class smartphone like the Motorola devices mentioned.

I'm open for any suggestions and hints as it would make development and testing quite a bit easier to not be limited to a single available (personal) device.

Additional Info:
I actually didn't bother to check for the Bluetooth sensor data as I don't intend to use it in any way and I didn't had our Bluetooth Beacons GUIDs at hand.

**Comment by bucurb** [comment]
Hi Tobias,
would you be able to share your accountId so we can track what's going on on the service side? Also can do a run and enable the logging and share the log - that it will definitely help.

**Comment by TobiasPott** [comment]
Hi @bucurb ,

sure I can, it'll take until tomorrow or early next week until I can run another test but I'll try to provide as much information as possible.
With the accountId you mean the id of the azure spatial anchor resource in our azure account?
And for the logging you refer to the on-device logging information (using logcat)?

Anything else that could help or information you may need?
Should I somehow provide the information you requested as PM or mail or post it here into the issue?

**Comment by bucurb** [comment]
Hi Tobias,
Yes, that would be the Accound Id (basically the GUID) that you would see in the Overview page of your Spatial Anchors Account in Azure. For logging you can either use logcat or our [diagnostics APIs](https://docs.microsoft.com/en-us/azure/spatial-anchors/concepts/logging-diagnostics), either to capture a diagnostics bundle or to redirect the information from the OnLogDebug and Error callbacks to your own log file. If you could send that in an email to me we can take it from there. Even the account Id would be useful initially, i.e. before you have the chance to capture client-side logs, and we can check if anything went wrong in the service.

**Comment by TobiasPott** [comment]
@bucurb I'll do the testing today and will initially create a logcat export of the logs.
If I managed to do another run with the diagnostics API enabled should I simply follow through with submitting the manifest as described in the linked documentation page (not sure how the association will be made to the specific device or whether or not I need to worry about it at all =))?

**Comment by bucurb** [comment]
I’d like to check a few more things if you have the time:
•	Uninstall the app and re-install it fresh. That should clear any permissions you might have accidentally declined and should prompt you again for everything it needs. 
•	Would it be possible to extract your app package and send us your manifest?
•	Can you alternatively try the Java version of the sample? Just in case Unity somehow messes-up the permissions it thinks it needs to prompt.

**Comment by bucurb** [comment]
I think we know what the problem is and are working on a fix. Will ping back once it's in.

**Comment by bucurb** [comment]
This should be fixed from 2.1.1. Could you give it another go please?

**Comment by TobiasPott** [comment]
Hi @bucurb 
I'll give it a try on Monday or Tuesday when I'm back in office.
I'm glad a fix for this has come that quick. Thank you in advance.

**Comment by TobiasPott** [comment]
Hi @bucurb ,
I finally found the time to update the SDK to 2.1.1 and run our application on the previously limited Motorola devices.
It now works as expected and on both devices (Motorola Moto G5S Plus and Moto G7 Play) can use GPS and Wifi sensor data.

As previously mentioned I'm not able to test the bluetooth feature due to the lack of access to beacons. But the reported problem that the stated devices cannot use the coarse localisation feature is solved in our case.

Thank you very much for the update. I'll consider this issue closed as it has been solved.
